modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -703,9 +724,104 @@ public Map<String, ModelsMap> postProcessAllModels(Map<String, ModelsMap> objs)
+                                if (!codegenModel.getImports().contains(imp)) {
+                                    inheritedImports.add(imp);
+                                    codegenModel.getImports().add(imp);
+                                }
+                            }
+                        }
+                    }
+                }
+                if (codegenModel.getParentModel() != null) {
+                    codegenModel.parentRequiredVars = new ArrayList<>(codegenModel.getParentModel().requiredVars);
+                }
+                // There must be a better way ...
+                for (String imp: inheritedImports) {
+                    String qimp = importMapping().get(imp);
+                    if (qimp != null) {
+                        Map<String,String> toAdd = new HashMap<>();
+                        toAdd.put("import", qimp);
+                        modelsAttrs.getImports().add(toAdd);
+                    }
+                }
+            }
+        }
+
+
+        if (isGeneratedConstructorWithAllArgs()) {
+            // compute allArgsConstructorVars to simplify the templates.
+            // It is used in the templates to generate the all arguments constructor.
+            // It should be equivalent to allVars. but with inheritanceSupport allVars does not match the specs.
+            for (CodegenModel cm : allModels.values()) {
+                List<Object> constructorArgs = new ArrayList<>();
+                cm.vendorExtensions.put("allArgsConstructorVars", constructorArgs);
+                if (sortModelPropertiesByInheritanceFirst) {
+                    constructorArgs.addAll(cm.parentVars);
+                    constructorArgs.addAll(cm.vars);
+                } else {
+                    constructorArgs.addAll(cm.vars);
+                    constructorArgs.addAll(cm.parentVars);
+                }
+                handleGenerateConstructorWithAllArgs(cm);
+            }
+        }
+
         return objs;
     }
 
+    private List<CodegenModel> getParentModelList(CodegenModel codegenModel) {
+        CodegenModel parentCodegenModel = codegenModel.parentModel;
+        List<CodegenModel> parentModelList = new ArrayList<>();
+        while (parentCodegenModel != null) {
+            parentModelList.add(parentCodegenModel);
+            parentCodegenModel = parentCodegenModel.parentModel;
+        }
+        if (sortModelPropertiesByInheritanceFirst) {
+            Collections.reverse(parentModelList);
+        }
+        return parentModelList;
+    }
+
+    /**
+     * trigger the generation of all arguments constructor or not.
+     *
+     * It avoids generating the same constructor twice.
+     */
+    protected void handleGenerateConstructorWithAllArgs(CodegenModel codegenModel) {
+        if (this.generatedConstructorWithAllArgs && !codegenModel.vars.isEmpty()) {
+            codegenModel.vendorExtensions.put("generatedConstructorWithAllArgs", true);
```

Comment:
why we need this extension?

shouldn't `generatedConstructorWithAllArgs` set via additionalProperties be enough?

Reply:
> why we need this extension?
> 
> shouldn't `generatedConstructorWithAllArgs` set via additionalProperties be enough?

We don't want two no arg constructors.
(one generated by the default template and one generated by the "all arg" constructor, the arg count being 0)

Reply:
what about checking `allVars` to see if it has zero argument or not?

Reply:
Maybe.

I had an issue with allVars and `supportsInheritance`.
`allVars` did not contain the parent vars.  I don't reproduce it anymore on master.
Is it something that was fixed recently?

I need to investigate more.

Reply:
allvars should contain parent vars

don't recall a recent fix.

Reply:
allVars was invalid for the contract issue_16797.yaml

See #18340

Reply:
replied to that. works for me in the latest master